### PR TITLE
fix,changes: change autocmd events to avoid cursor flashing

### LINKED ISSUE #82: The `changes` component makes my cursor blink every time I move it
If I comment out the `changes` component in my config it stops happening, so it's coming from there. I'll investigate and see if I can do a PR.

### PR DESCRIPTION
**Pull request by mrjones2014**
Fixes #82 

These autocmd events I think accomplish the same thing you were trying to do.

### CODE REVIEW

File: lua/ide/components/changes/component.lua [hunk 1]
```diff
@@ -368,7 +368,7 @@ ChangesComponent.new = function(name, config)
         return commands.new(self).get()
     end
 
-    vim.api.nvim_create_autocmd({ "BufEnter", "CursorHold" },
+    vim.api.nvim_create_autocmd({ "BufEnter", "TextChanged", "InsertLeave" },
```
Comment (ldelossa) [inline lua/ide/components/changes/component.lua:371]: Hey there. I originally had this as text changed but i remeber it causing some issues. Let me see if i still can determine why

Comment (mrjones2014) [reply lua/ide/components/changes/component.lua:371]: `TextChanged` only fires from normal mode, you can include insert mode with `TextChangedI`

Comment (ldelossa) [reply lua/ide/components/changes/component.lua:371]: No it was a different issue, doing the lines with the plugin itself. I'll test

**Comment by ldelossa** [comment]
Hey, can you update commit message like other? 

"fix,changes:"

**Comment by ldelossa** [comment]
This change seems odd to me. We use CursorHold quite often, why would this component be the only one to cause cursor blink

**Comment by mrjones2014** [comment]
Probably has something to do with `api-fast`

**Comment by mrjones2014** [comment]
> Hey, can you update commit message like other?
> 
> "fix,changes:"

Done.

**Comment by ldelossa** [comment]
Hey, I am testing this now, it does not work well at all. 

Basically, you wind up needing to do an "insert" and "normal" mode cycle before the status component updates, after saving the changes to a file.

**Comment by mrjones2014** [comment]
We could use `BufWritePost` maybe?

**Comment by ldelossa** [comment]
Are you sure this is not terminal related? The Cursor is usually an element of the terminal, not neovim.

**Comment by mrjones2014** [comment]
Yes I'm sure. I'm able to reproduce in multiple terminal emulators, and only while this specific autocmd is unfixed.

>The Cursor is usually an element of the terminal, not neovim.

I don't think that's correct.

**Review by ldelossa — APPROVED** [review]

**Comment by ldelossa** [comment]
Ah, its not going to work for file deletions too well. You won't see the removed deleted files until you change buffers.

**Comment by mrjones2014** [comment]
Maybe we can just have it poll based on a timer and not use an autocmd at all? 

Something like this will check every 2  seconds:

```lua
local function check()
  -- do the git updates

  vim.defer_fn(check, 2000)
end

vim.defer_fn(check, 2000)
```

**Comment by ldelossa** [comment]
Going to revert this for now. But lets keep brain storming.

**Comment by ldelossa** [comment]
May use vim.loop to watch the .git directory, and register a callback on fs change.

**Comment by mrjones2014** [comment]
Let's take a look at how `gitsigns.nvim` does it. It's extremely responsive and never blocks anything.

**Comment by mrjones2014** [comment]
Looks like they are using a git directory watcher via `vim.loop.new_fs_foll(true):start(git_dir, interval, callback)`

https://github.com/lewis6991/gitsigns.nvim/blob/main/teal/gitsigns/manager.tl#L361

**Comment by ldelossa** [comment]
However to do this right, the workspace controller should create this, and allow for components to 'register' callbacks on git events. 

This will take a bit (not much) planning to do, probably in a PR of its own.

**Comment by mrjones2014** [comment]
Please tag me on the PR! I'm going to continue using my fork with this PR in place because its very frustrating to use without it, the cursor blinks drive me nuts.

**Comment by ldelossa** [comment]
I'm assuming you don't have your terminal setting to blink the cursor always? This is probably why I do not see or get bothered by it.

**Comment by mrjones2014** [comment]
Nope, my terminal cursor never blinks. In Neovim, without this fix, it briefly flashes any time I move the cursor in normal mode.

**Pull request merged**